core/src/test/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolverTests.java

```diff
@@ -642,6 +642,7 @@ public void testConcreteIndicesWildcardWithNegation() {
 
         indexNames = indexNameExpressionResolver.concreteIndexNames(state, IndicesOptions.lenientExpandOpen(), "*", "-*");
         assertEquals(0, indexNames.length);
+        assertWarnings("use of + is deprecated in index names as it is implicit");
```

Comment:
can you add a specific test method that checks the new deprecation line and remove usages of '+' elsewhere?